static/js/src/advantage/react/components/Subscriptions/ExpiryNotification/ExpiryNotification.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { mount } from "enzyme";
+
+import ExpiryNotification, {
+  ExpiryNotificationSize,
+} from "./ExpiryNotification";
+import { userSubscriptionStatusesFactory } from "advantage/tests/factories/api";
+import { Notification } from "@canonical/react-components";
+
+describe("ExpiryNotification", () => {
+  it("can display an expiring notification", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_expiring: true,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(wrapper.find("[data-test='is_expiring-large']").exists()).toBe(true);
+  });
+
+  it("can display a large expired notification", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_expired: true,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(wrapper.find("[data-test='is_expired-large']").exists()).toBe(true);
+  });
```

Comment:
I think we'd get more value from this test if we were testing results that the user actually cares about instead (actual text that's displayed).

What you're testing here is that your logic for generatic test ids is functioning properly. It's possible for this test to pass even if the large/small notification isn't displayed correctly.

Parts that are probably more important than a test ID aren't covered (like displaying a title for large notifications).

Reply:
Thanks I really did write some terrible tests here. I reckon this is a tricky one to find a balance between writing maintainable tests and testing what we're actually producing. I'll see what I can do to fix these up.